trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/VLM.py

```diff
@@ -140,131 +181,244 @@ def VLM(conditions,settings,geometry):
+    ZBH = VD.ZBH    
+    YBH = VD.YBH*1.
+    
+    XA1 = VD.XA1*1.
+    XB1 = VD.XB1*1.
+    YA1 = VD.YA1
+    YB1 = VD.YB1    
+    ZA1 = VD.ZA1
+    ZB1 = VD.ZB1 
+    
+    XA2 = VD.XA2
+    XB2 = VD.XB2
+    YA2 = VD.YA2
+    YB2 = VD.YB2    
+    ZA2 = VD.ZA2
+    ZB2 = VD.ZB2 
+    
+    XCH = VD.XCH
+    YCH = VD.YCH
+    ZCH = VD.ZCH
+    
+    XA_TE =  VD.XA_TE
+    XB_TE =  VD.XB_TE
+    YA_TE =  VD.YA_TE
+    YB_TE =  VD.YB_TE
+    ZA_TE =  VD.ZA_TE
+    ZB_TE =  VD.ZB_TE     
+
+    # additional VD preprocessing
+    # from here on, VD will also be used to hold some processed information about geometry
+    # for the easier passage of this information into functions
+    XBAR    = np.ones(sum(LE_ind)) * x_m
+    ZBAR    = np.ones(sum(LE_ind)) * z_m
+    
+    X1c   = (XA1+XB1)/2
+    X2c   = (XA2+XB2)/2
+    Z1c   = (ZA1+ZB1)/2
+    Z2c   = (ZA2+ZB2)/2
+    SLOPE = (Z2c - Z1c)/(X2c - X1c)
+    SLE   = SLOPE[LE_ind]
+    
+    VD.XBAR  = XBAR * 1
+    VD.ZBAR  = ZBAR * 1
+    VD.SLOPE = SLOPE*1
+    VD.SLE   = SLE*1
+    
+    # ---------------------------------------------------------------------------------------
+    # STEP 10: Generate A and RHS matrices from VD and geometry
+    # ------------------ --------------------------------------------------------------------    
     # Compute flow tangency conditions
     phi   = np.arctan((VD.ZBC - VD.ZAC)/(VD.YBC - VD.YAC))*ones # dihedral angle 
     delta = np.arctan((VD.ZC - VD.ZCH)/((VD.XC - VD.XCH)*ones)) # mean camber surface angle 
 
-    # Build the vector 
-
-    RHS  ,Vx_ind_total , Vz_ind_total , V_distribution , dt = compute_RHS_matrix(delta,phi,conditions,geometry,\
-                                                                                 pwm,bemt_wake,ito,wdt,nts )    
+    # Build the RHS vector    
+    rhs = compute_RHS_matrix(delta,phi,conditions,settings,geometry,pwm,bemt_wake,ito,wdt,nts ) 
+    RHS     = rhs.RHS*1
+    ONSET   = rhs.ONSET*1
 
     
     # Build induced velocity matrix, C_mn
     # This is not affected by AoA, so we can use unique mach numbers only
     m_unique, inv = np.unique(mach,return_inverse=True)
     m_unique      = np.atleast_2d(m_unique).T
-    C_mn_small, s, RFLAG_small = compute_wing_induced_velocity(VD,m_unique)
+    C_mn_small, s, RFLAG_small, EW_small = compute_wing_induced_velocity(VD,m_unique,computed_in_VLM=True)
     
     C_mn  = C_mn_small[inv,:,:,:]
     RFLAG = RFLAG_small[inv,:]
+    EW    = EW_small[inv,:,:]
 
     # Turn off sonic vortices when Mach>1
     RHS = RHS*RFLAG
     
     # Build Aerodynamic Influence Coefficient Matrix
-    A =   np.multiply(C_mn[:,:,:,0],np.atleast_3d(np.sin(delta)*np.cos(phi))) \
-        + np.multiply(C_mn[:,:,:,1],np.atleast_3d(np.cos(delta)*np.sin(phi))) \
-        - np.multiply(C_mn[:,:,:,2],np.atleast_3d(np.cos(phi)*np.cos(delta)))   # validated from book eqn 7.42      
+    use_VORLAX_induced_velocity = getattr(settings, 'use_VORLAX_matrix_calculation', False)
```

Comment:
why this convoluted syntax for a simple boolean?

Default it and unpack at the top.

Reply:
right my bad that's code that just didn't get updated after I added the use_VORLAX_matrix_calculation attribute to Vortex_Lattice().settings